packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,17 +6,87 @@
  */
 
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, isSFDXContainerMode } from '../util';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+type CLIOrgData = {
+  status: number;
+  result: {
+    orgId: string;
+    url: string;
+    username: string;
+  };
+};
+
+class ForceOrgOpenContainerExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
+      .withArg('force:org:open')
+      .withLogName('force_org_open_default_scratch_org')
+      .withArg('--urlonly')
+      .withJson()
+      .build();
+  }
+
+  private buildUserMessageWith(orgData: CLIOrgData): string {
+    return nls.localize(
+      'force_org_open_container_mode_message_text',
+      orgData.result.orgId,
+      orgData.result.username,
+      orgData.result.url
+    );
+  }
+
+  private openBrowserWithCliResponse(cliResponseJSON: string | Buffer) {
+    try {
+      const cliOrgData: CLIOrgData = JSON.parse(cliResponseJSON.toString());
+      const authenticatedOrgUrl: string = cliOrgData.result.url;
+
+      channelService.appendLine(this.buildUserMessageWith(cliOrgData));
+
+      if (authenticatedOrgUrl) {
+        vscode.env.openExternal(vscode.Uri.parse(authenticatedOrgUrl));
+      }
+    } catch (e) {
+      telemetryService.sendErrorEvent(e.message, e.stack);
+    }
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(() => {
+      this.logMetric(execution.command.logName, startTime);
+    });
+    execution.stdoutSubject.subscribe(cliResponseJSON =>
```

Comment:
The CLI returns a stream with the response, we normally use `execution.stdoutSubject.subscribe` to collect the data into a string variable and later process it in `execute.processExitSubject.subscribe`. You can take a look at how we do that in `forceOrgCreate.ts` and other commands that implement their own `execute` method.